### PR TITLE
[go1.15] Update to go1.15

### PR DESCRIPTION
**Pull request by justaugustus**
**What type of PR is this?**

/kind cleanup
/area dependency release-eng
/sig release

**What this PR does / why we need it**:

**IT'S HAPPENING :tada:**
Update to go1.15.0

Signed-off-by: Stephen Augustus <saugustus@vmware.com>

cc: @kubernetes/release-engineering @kubernetes/release-team-leads @kubernetes/sig-release-admins @kubernetes/sig-scalability @liggitt @dims 
Tracking issue: https://github.com/kubernetes/release/issues/1421

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
- Kubernetes is now built with go1.15.0
```

### CODE REVIEW

File: test/images/Makefile [hunk 1]
```diff
@@ -16,7 +16,7 @@ REGISTRY ?= gcr.io/kubernetes-e2e-test-images
 GOARM ?= 7
 DOCKER_CERT_BASE_PATH ?=
 QEMUVERSION=v2.9.1
-GOLANG_VERSION=1.15rc2
+GOLANG_VERSION=1.15.0
```
Comment (dims) [inline test/images/Makefile:19]: is this just `1.15` or is it `1.15.0`?

File: build/root/WORKSPACE [hunk 2]
```diff
@@ -24,10 +23,7 @@ load("@io_k8s_repo_infra//:repos.bzl", repo_infra_configure = "configure", repo_
 #            'override_go_version': used to specify an alternate go version provided
 #                                   by kubernetes/repo-infra
 repo_infra_configure(
-    # TODO(go1.15): Uncomment once we've published a repo-infra release that supports go1.15
-    #go_version = "1.15.0",
-    # TODO(go1.15): Comment out once we've published a repo-infra release that supports go1.15
-    override_go_version = "1.15.0-rc.2",
+    go_version = "1.15.0",
```
Comment (liggitt) [inline build/root/WORKSPACE:26]: > unknown Go version: 1.15.0

should this be "1.15"?

Comment (liggitt) [reply build/root/WORKSPACE:26]: this missing ".0" really bugs me

Comment (BenTheElder) [reply build/root/WORKSPACE:26]: yeah :|
it's `1.15`

**Comment by BenTheElder** [comment]
don't forget to update the release notes too! go doesn't have .0 versions :roll_eyes: 
> - Kubernetes is now built with go1.15.0

**Comment by justaugustus** [comment]
Yep, it's throwing errors because of the .0. I'm fixing up regexes locally :)

File: build/root/WORKSPACE (hunk 2 above)
Comment (justaugustus) [reply build/root/WORKSPACE]: It really bugs me too.

File: build/dependencies.yaml [hunk 3]
```diff
@@ -82,21 +82,17 @@ dependencies:
...
+    version: 1.15
     refPaths:
     - path: build/build-image/cross/VERSION
     - path: build/root/WORKSPACE
-      match: (override_)?go_version = "((([0-9]+)\.([0-9]+)\.([0-9]+)(?:-([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?)(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?)"
-
-  - name: "golang: upstream container image"
-    version: 1.15rc2
-    refPaths:
+      match: (override_)?go_version = "\d+.\d+(alpha|beta|rc)?\.?\d+"
```
Comment (liggitt) [inline build/dependencies.yaml:90]: I think you want something like:
```
"\d+\.\d+(alpha\d*|beta\d*|rc\d*|\.\d+)?"
```

```
# one or more digits, followed by .
\d+\.
# one or more digits
\d+
# optionally followed by alpha/beta/rc of any number, or a specific patch release
(
  alpha\d*
  |
  beta\d*
  |
  rc\d*
  |
  \.\d+
)?
```

Comment (liggitt) [reply build/dependencies.yaml:90]: follow up for this is fine, just noting that the only reason this currently passes 1.15 is:

```
"\d+.\d+(alpha|beta|rc)?\.?\d+"
"1  .1                     5"
```

and the `.` in the regex would match any character, not just a literal `.`

File: test/images/Makefile (hunk 1 above)
Comment (liggitt) [reply test/images/Makefile:19]: the docker image includes the `.0` 😜 

https://hub.docker.com/_/golang?tab=tags&page=1&name=1.15.0

**Comment by liggitt** [comment]
/lgtm
/approve

followups:
* switch CI jobs to 1.15.0
* set `minimum_go_version=go1.15.0` in golang.sh

File: build/dependencies.yaml (hunk 3 above)
Comment (justaugustus) [reply build/dependencies.yaml:90]: Yep, will issue a follow up to clean up some of the regex in `dependencies.yaml`.
It's a bit Wild Wild West in there at this point.

File: build/root/WORKSPACE (hunk 2 above)
Comment (dims) [reply build/root/WORKSPACE]: w00t! i caught something in my review.... now make me an approver :)

**Comment by liggitt** [comment]
> * set `minimum_go_version=go1.15.0` in golang.sh

opened in https://github.com/kubernetes/kubernetes/pull/93969

**Pull request merged**